Loop/Managers/RemoteDataServicesManager.swift

```diff
@@ -176,10 +176,6 @@ final class RemoteDataServicesManager {
         clearSettingsQueryAnchor(for: remoteDataService)
     }
 
```

Comment:
Remove this in favor of `triggerBackgroundUpload` above and `triggerUpload` which offers a callback on completion.

There is a chance the uploadGroup will include other things unrelated to what we triggered, but I think that is ok as we had this issue before with `waitForUploadsToFinish`

Reply:
My one concern about the background task strategy, rather than using the time that iOS has allocated us for processing the remote command (`didReceiveRemoteNotification`), is that we're depleting the limited resource of *background task* time.  We currently don't use background tasks for much else, so there's not currently a lot of contention. But it is a limited resource, and if we could take advantage of the time allocated for `didReceiveRemoteNotification`, that would be more aligned with the expectations of iOS.

Reply:
Just to confirm before I take this direction - I can easily switch to using the method that was used previously here for the remote uploads `waitForUploadsToFinish`.

To eliminate my additional background tasks for `self.enactBolus` `carbStore.addCarbEntry`, I assume you'd like a similar strategy followed -- using dispatch groups and waiting on their result rather than a background task. That will add some new Xcode thread inversion warnings, just FYI.

I can go  that direction as there are tradeoffs either way but I just want to clarify that's a change you'd like to see.